### PR TITLE
Lps 148397 Allow to download Folder as .7z

### PR DESCRIPTION
**Pull request by ces-hongvo**
Hi @tototrinh 
The original issue 
https://issues.liferay.com/browse/LPP-44132
https://issues.liferay.com/browse/LPS-148397
It's suggested to create 7z download in the PTR https://issues.liferay.com/browse/PTR-2903

Please review this PR and leave your comments.
Thank you

### CODE REVIEW

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/FolderActionDisplayContext.java [hunk 1]
```diff
@@ -207,18 +207,12 @@ public String getDeleteFolderURL() throws PortalException {
...
-			_dlRequestHelper.getLiferayPortletResponse();
-
-		ResourceURL resourceURL = liferayPortletResponse.createResourceURL();
-
-		resourceURL.setParameter("folderId", String.valueOf(_getFolderId()));
-		resourceURL.setParameter(
-			"repositoryId", String.valueOf(_getRepositoryId()));
-		resourceURL.setResourceID("/document_library/download_folder");
+	public String getDownloadFolderSevenZipURL() {
+		return _getDownloadFolderURL("7z");
```
Comment (tototrinh) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/FolderActionDisplayContext.java:211]: _.7z_ as a parameter would be better?

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/FolderActionDisplayContext.java [hunk 2]
```diff
@@ -207,18 +207,12 @@ public String getDeleteFolderURL() throws PortalException {
...
+	}
 
-		return resourceURL.toString();
+	public String getDownloadFolderURL() {
+		return _getDownloadFolderURL("zip");
```
Comment (tototrinh) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/FolderActionDisplayContext.java:215]: same above.

File: portal-impl/src/com/liferay/portal/zip/SevenZipWriterImpl.java [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+		if (ExportImportThreadLocal.isExportInProcess()) {
+			addEntry(name, StreamUtil.toByteArray(inputStream));
+
+			return;
+		}
+
+		FileSystem fileSystem = _path.getFileSystem();
+
+		Path path = fileSystem.getPath(
```
Comment (tototrinh) [inline portal-impl/src/com/liferay/portal/zip/SevenZipWriterImpl.java:108]: This is the path of the file entry? `path` and `_path` make me feel confused. What do you think about changing it?

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/FolderActionDisplayContext.java (hunk 1 above)
Comment (hongvo2308) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/FolderActionDisplayContext.java]: I still think "7z" is better. Please explain why ".7z" is better for you?

**Comment by ces-hongvo** [comment]
@tototrinh 
I have updated the PR. Please review it again and leave your comments.
Thank you.

**Comment by ces-totrinh** [comment]
It looks good to me. Thanks.

**Comment by ces-totrinh** [comment]
Forwarded here: liferay-ts-reviewer#27

**Pull request closed without merging**